### PR TITLE
[TypeDeclaration] AddVoidReturnTypeWhereNoReturnRector turns `@return never` into less specific `@return void`

### LINKED ISSUE #6892: AddVoidReturnTypeWhereNoReturnRector turns `@return never` into less specific `@return void`
# Bug Report

https://getrector.org/demo/1ec63f14-e7e6-6e54-8cd7-8dbf2eede3d2

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/rectorphp/rector/issues/6892

### CODE REVIEW

File: rules-tests/TypeDeclaration/Rector/ClassMethod/AddVoidReturnTypeWhereNoReturnRector/FixtureUsePhpDoc/never-types.php.inc [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    /**
+     * @return no-return
+     */
+    public function noReturn()
+    {
+        header('Location: example.com');
+        exit();
+    }
```
Comment (staabm) [inline rules-tests/TypeDeclaration/Rector/ClassMethod/AddVoidReturnTypeWhereNoReturnRector/FixtureUsePhpDoc/never-types.php.inc:41]: it seems rector does not support all "aliases" of the bottom-type phpstan supports:

https://phpstan.org/writing-php-code/phpdoc-types#bottom-type

only `@return never` is turned into the `NeverType` ..

Comment (staabm) [reply rules-tests/TypeDeclaration/Rector/ClassMethod/AddVoidReturnTypeWhereNoReturnRector/FixtureUsePhpDoc/never-types.php.inc:41]: fixed in the latest commit

File: packages/StaticTypeMapper/Mapper/ScalarStringToTypeMapper.php [hunk 2]
```diff
@@ -36,7 +36,7 @@ final class ScalarStringToTypeMapper
         ResourceType::class => ['resource'],
         CallableType::class => ['callback', 'callable'],
         ObjectWithoutClassType::class => ['object'],
-        NeverType::class => ['never'],
+        NeverType::class => ['never', 'never-return', 'never-returns', 'no-return'],
```
Comment (staabm) [inline packages/StaticTypeMapper/Mapper/ScalarStringToTypeMapper.php:39]: added all phpstan bottom types: see https://phpstan.org/writing-php-code/phpdoc-types#bottom-type

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Thank you @staabm

**Pull request merged**